Mathlib/Tactic/FunProp/Core.lean

```diff
@@ -20,6 +20,18 @@ open Lean Meta Qq
 
 namespace Meta.FunProp
 
+
+/-- Puts function `f` into expected form
+  1. has at least one lambda binder
+  2. body of the lambda is (morphism)beta reduced -/
```

Comment:
I'm getting that impression from your change :)

Precondition: `f` is a function (do nothing otherwise)
Postcondition: point 1 and 2

I will change it to
```
Puts function `f` into expected form, that means
1. ...
2. ...

Does nothing if `f` is not a function.
```

Reply:
I like your new version as well, thanks!